languages/wikipron/src/codes.py

```diff
@@ -0,0 +1,174 @@
+to be added manually as well.
+"""
+
+import requests
+import requests_html
+import re
+import csv
+import json
+
+
+def cat_info(cat_title):
+    cat_info_params = {
+        "action": "query",
+        "format": "json",
+        "titles": cat_title,
+        "prop": "categoryinfo"
+    }
+    data = requests.get(
+        "https://en.wiktionary.org/w/api.php?",
+        params=cat_info_params
+    ).json()
+    pages = data["query"]["pages"]
+
+    for k, v in pages.items():
+        num_of_pages = v["categoryinfo"]["pages"]
+        if num_of_pages >= 100:
+            isolate_language_category = re.search(
+                r"(\s+)terms(\s+)",
+                v["title"]
+            )
+            yield (
+                v["title"][0:isolate_language_category.start()],
+                num_of_pages
+            )
+        else:
+            continue
+
+
+def cat_members():
+    cat_member_params = {
+        "action": "query",
+        "cmtitle": "Category:Terms with IPA pronunciation by language",
+        "cmlimit": "500",
+        "list": "categorymembers",
+        "format": "json"
+    }
+    while True:
+        data = requests.get(
+            "https://en.wiktionary.org/w/api.php?",
+            params=cat_member_params
+        ).json()
+        for member in data["query"]["categorymembers"]:
+            yield from cat_info(member["title"])
+        if "continue" not in data:
+            break
+        continue_code = data["continue"]["cmcontinue"]
+        cat_member_params["cmcontinue"] = continue_code
+
+
+def scrape_wiktionary_info(lang_title):
+    name = None
+    code = None
+
+    session = requests_html.HTMLSession()
+    language_page = session.get(
+        f"https://en.wiktionary.org/wiki/{lang_title}_language",
+        timeout=10
+    )
+    lang_table = language_page.html.find(
+        '.language-category-info > tbody > tr'
+    )
+
+    i = 0
+    while i < len(lang_table):
+        if "Canonical name" in lang_table[i].text:
+            name = lang_table[i].text
+            name = name.split('\n')[1]
+        elif "Language code" in lang_table[i].text:
+            # Canonical name (should) always be filled already.
+            # We grab Wiktionary language code as our entry point
+            # to compare/confirm with ISO 639 TSV File because
+            # Wiktionary name does not always correspond with ISO language name
+            # Wiktionary: Ancient Greek; ISO: Greek, Ancient (to 1453)
+            code = lang_table[i].text
+            code = code.split('\n')[1]
+            break
+        i += 1
+    return name, code
+
+
+def main():
+    new_languages = {}
+    failed_languages = {}
+    prev_languages_file = open("languages.json", "r")
+    prev_languages = json.load(prev_languages_file)
+    language_codes_file = open("iso-639-3_20190408.tsv", "r")
+    iso_list = csv.reader(language_codes_file, delimiter="\t")
+
+    for lang_page_title, total_pages in cat_members():
+        lang = {}
```

Comment:
It'd be nice to add a `print` line (as simple as `print(lang_page_title)`, for instance) at the beginning of the iteration, so that we see the script is chugging along.